force-app/main/default/classes/FlowControlLoopsCollections.cls

```diff
@@ -83,8 +100,16 @@ public with sharing class FlowControlLoopsCollections {
      * @return A string indicating whether the number is "Even" or "Odd", or null if the number is null.
      */
     public static String checkEvenOdd(Integer a) {
-        return null; // Replace null with the variable you used to store the result
-    }
+        if (a == null) {
+            return null;
+        } else {
+            if (((a / 2) *2) == a) {
+                return 'Even';
+            } else {
+                return 'Odd';
+            }
+        }
+        }
```

Comment:
This is really hard to read. Try to avoid doing if statements within if statements. You are already calling `return null` in the first if statement, so you don't need to do `else` or call `if` again. you can just do your logic after that, no need to put it inside else.

Look into the `Math.mod` to do Modulus operations.